common/djangoapps/third_party_auth/signals/tests/test_handlers.py

```diff
@@ -33,100 +33,77 @@ def setUp(self):
+            )
+
     @mock.patch('common.djangoapps.third_party_auth.signals.handlers.set_custom_attribute')
-    def test_saml_config_signal_handlers(
-            self, mock_set_custom_attribute, enabled, simulate_error,
-            description, expected_calls, expected_call_count, error_message=None):
+    def test_saml_config_signal_handlers_with_error(self, mock_set_custom_attribute):
         """
-        Test SAML configuration signal handlers under different conditions.
+        Test error handling when signal handlers encounter an exception.
+
+        Verifies that error information is properly captured when provider updates fail.
         """
-        with override_settings(ENABLE_SAML_CONFIG_SIGNAL_HANDLERS=enabled):
-            if simulate_error:
-                # Simulate an exception in the provider config update logic
-                with mock.patch(
-                    'common.djangoapps.third_party_auth.models.SAMLProviderConfig.objects.current_set',
-                    side_effect=Exception(error_message)
-                ):
-                    self.saml_config.entity_id = 'https://updated.example.com'
-                    self.saml_config.save()
-            else:
+        error_message = "Test error"
+        with override_settings(ENABLE_SAML_CONFIG_SIGNAL_HANDLERS=True):
+            # Simulate an exception in the provider config update logic
+            with mock.patch(
+                'common.djangoapps.third_party_auth.models.SAMLProviderConfig.objects.current_set',
+                side_effect=Exception(error_message)
+            ):
                 self.saml_config.entity_id = 'https://updated.example.com'
                 self.saml_config.save()
 
-            expected_calls_with_id = []
-            for call_obj in expected_calls:
-                args = list(call_obj[1])
-                if args[1] == 'CONFIG_ID':
-                    args[1] = self.saml_config.id
-                expected_calls_with_id.append(call(args[0], args[1]))
+            expected_calls = [
+                call('saml_config_signal.enabled', True),
+                call('saml_config_signal.new_config_id', self.saml_config.id),
+                call('saml_config_signal.slug', 'test-config'),
+            ]
 
             # Verify expected calls were made
-            mock_set_custom_attribute.assert_has_calls(expected_calls_with_id, any_order=False)
+            mock_set_custom_attribute.assert_has_calls(expected_calls, any_order=False)
 
-            # Verify total call count
-            assert mock_set_custom_attribute.call_count == expected_call_count, (
-                f"Expected {expected_call_count} calls for {description}, "
+            # Verify total call count (includes error_message call)
+            assert mock_set_custom_attribute.call_count == 4, (
+                f"Expected 4 calls for error case, "
                 f"got {mock_set_custom_attribute.call_count}"
             )
```

Comment:
Same cleanup as above.
```suggestion
            assert mock_set_custom_attribute.call_count == 4
```